immutable_tree.go

```diff
@@ -109,12 +109,12 @@ func (t *ImmutableTree) Version() int64 {
 	return t.version
 }
 
-// Height returns the height of the tree.
+// Height returns the depth of the tree.
```

Comment:
Can we rename this to `SubtreeHeight()` and update the docs? The docs seem to say that this returns depth